dcpy/models/product/metadata.py

```diff
@@ -71,3 +85,52 @@ def get_tagged_destinations(self, tag) -> dict[str, dict[str, DatasetMetadata]]:
                 if tag in dest.tags:
                     found_tagged_dests[ds.id][dest.id] = ds
         return found_tagged_dests
+
+    def validate_dataset_metadata(self) -> dict[str, list[str]]:
+        md = self.get_product_metadata()
+        product_errors = {}
+
+        for ds_id in md.datasets:
+            errors = []
+            try:
+                errors = self.get_product_dataset(ds_id, md).validate_consistency()
+            except Exception as e:
+                errors = [
+                    f"Error instantiating dataset metadata for {md.id}: {ds_id}: {e}"
+                ]
+            if errors:
+                product_errors[ds_id] = errors
+        return product_errors
+
+
+class MetadataRepoFolder(SortedSerializedBase, extra="forbid"):
```

Comment:
just wondering - again, is this also "org"-level? Would you prefer this to be defined as a "Repo" as it is now, or as an "Org" which is tightly coupled to a repo. Good either way, just curious on your thinking

Reply:
Good question - so what I was shooting for was something very literal, to indicate that it's a class representation for the top-level folder of the metadata repo. But since it has template_vars, it's not quite so literal. Changed to OrgMetadata